frontend/javascripts/dashboard/dataset/dataset_settings_view.tsx

```diff
@@ -243,68 +151,12 @@ class DatasetSettingsView extends React.PureComponent<PropsWithFormAndRouter, St
         isLoading: true,
       });
       let dataset = await getDataset(this.props.datasetId);
-      let dataSource;
-
-      const dataSourceSettingsStatus = {
-        appliedSuggestions: AppliedSuggestionsEnum.NoAvailableSuggestions,
-        isJSONFormatValid: IsJSONFormatValidEnum.No,
-      };
-      const {
-        dataSource: inferredDataSource,
-        messages: dataSourceMessages,
-        previousDataSource: savedDataSourceOnServer,
-      } = await getDatasetDatasource(dataset);
-      const didParsingTheSavedDataSourceJSONSucceed =
-        savedDataSourceOnServer != null && savedDataSourceOnServer.status == null;
+      const dataSource = await readDatasetDatasource(dataset);
 
       // Ensure that zarr layers (which aren't inferred by the back-end) are still
       // included in the inferred data source
-      if (
-        savedDataSourceOnServer &&
-        "dataLayers" in savedDataSourceOnServer &&
-        inferredDataSource != null
-      ) {
```

Comment:
This cannot be done as there is no longer a `inferredDataSource`. So does this maybe result in errors with zarr layers?